fenix/app/src/main/java/org/mozilla/fenix/settings/search/SearchEngineShortcuts.kt

```diff
@@ -0,0 +1,254 @@
+import androidx.compose.material.IconButton
+import androidx.compose.material.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.MutableState
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.asImageBitmap
+import androidx.compose.ui.res.painterResource
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.text.font.FontWeight
+import androidx.compose.ui.unit.DpOffset
+import androidx.compose.ui.unit.dp
+import mozilla.components.browser.state.search.SearchEngine
+import mozilla.components.browser.state.state.SearchState
+import mozilla.components.browser.state.state.availableSearchEngines
+import mozilla.components.browser.state.store.BrowserStore
+import mozilla.components.lib.state.ext.observeAsComposableState
+import org.mozilla.fenix.R
+import org.mozilla.fenix.compose.ContextualMenu
+import org.mozilla.fenix.compose.MenuItem
+import org.mozilla.fenix.theme.FirefoxTheme
+
+/**
+ * Top-level UI for search shortcuts settings.
+ */
+@Suppress("LongParameterList")
+@Composable
+fun SearchEngineShortcuts(
+    categoryTitle: String,
+    store: BrowserStore,
+    onCheckboxClicked: (SearchEngine, Boolean) -> Unit,
+    onEditEngineClicked: (SearchEngine) -> Unit,
+    onDeleteEngineClicked: (SearchEngine) -> Unit,
+    onAddEngineClicked: () -> Unit,
+) {
+    val searchState = store.observeAsComposableState { it.search }.value ?: SearchState()
+    val searchEngines = with(searchState) {
+        regionSearchEngines + additionalSearchEngines + availableSearchEngines + customSearchEngines
+    }
+    val disabledShortcutsIds = searchState.disabledSearchEngineIds
+
+    LazyColumn(
+        modifier = Modifier
+            .background(color = FirefoxTheme.colors.layer1)
+            .fillMaxSize(),
+    ) {
+        item {
+            Title(title = categoryTitle)
+
+            Spacer(modifier = Modifier.height(12.dp))
+        }
+
+        items(
+            items = searchEngines,
+            key = { engine -> engine.id },
+        ) {
+            SearchItem(
+                engine = it,
+                name = it.name,
+                isEnabled = !disabledShortcutsIds.contains(it.id),
+                onCheckboxClicked = onCheckboxClicked,
+                onEditEngineClicked = onEditEngineClicked,
+                onDeleteEngineClicked = onDeleteEngineClicked,
+            )
+        }
+
+        item {
+            AddEngineButton(onAddEngineClicked = onAddEngineClicked)
+        }
+    }
+}
+
+@Composable
+private fun Title(title: String) {
+    Box(
+        modifier = Modifier
+            .height(48.dp)
+            .fillMaxWidth(),
+        contentAlignment = Alignment.CenterStart,
+    ) {
+        Text(
+            text = title,
+            color = FirefoxTheme.colors.textAccent,
+            fontWeight = FontWeight.W400,
```

Comment:
> hey, @gabrielluong, thanks for the early feedback, I have addressed it (except the "Go to main component" thing, editor mode isn't available for me).
> 
> I _really_ want to land it as soon as possible and ship unified-search in 115, that's the only blocker. And, generally speaking, I don't think nits should be blocking a feature. Let's do things like extracting a general compose component to AC layer as follow ups, and now focus only on blocking changes.
> 
> Once we uplift it to beta and gain confidence, I will be doing some refactoring – removing duplicated setting screen, etc. It looks like a good opportunity to extract a basic `@Compose `component as well. Writing it now doesn't really help with landing the feature and is more of scope creeping.
> 
> I wanted to hide the new UI behind the feature flag, so I copy-pasted a lot of existing code with slight modifications to match the new UI to provide an alternative path for up to show the settings screen. Changing initial designed would make the patch smaller, but more difficult to roll-back.

Could you ask @MozillaNoah for the UI review since Jon only approved the non-UI bits? I don't have the bandwidth for this. As any code reviewer, the only takeaway here is that we don't necessarily want to do +1189 lines of code review when we could break it down and land incrementally, and to be fair, it's harder to give constructive feedback when there's just so many things to comment on with any large PR.

I have 2 areas that I would ask that we focus on:
- Making sure we are aligned with the figma designs. In the highlighted code here, I found that the title is using W500 that already exists in the headline8 text style. ([figma for the title](https://www.figma.com/file/WzMZduDsU0dJUj8P3gqiPf/Search-engine-settings?type=design&node-id=40-22583&t=BAdltawcpOKBZbfs-0)). 
- Addressing the existing feedback. I would suggest resolving the existing review comments with a comment about the actions taken or decision made so that we're closing the loop. I know you said you addressed all the feedback, but it's kinda hard to tell since I could still see paddings with only "start" instead of "horizontal". If the changes were intentional, then a comment would clarify that. Even if the comment only says that you would file a follow up.

Reply:
@gabrielluong 
I did, but he didn't have time. Somebody else will take a look, no worries.

This design has some inconsistencies with the rest of the app that I decided not to introduce. `PreferenceCategory` class is using 400, and it is very obsious coming from the [searching_preference](https://searchfox.org/mozilla-mobile/source/firefox-android/fenix/app/src/main/res/xml/search_preferences.xml) screen.
 400 | 500
 --- | --- 
<img src="https://github.com/mozilla-mobile/firefox-android/assets/92760693/a7f7dc67-018a-4cd8-a33f-6347d1052198" width="300"> |  <img src="https://github.com/mozilla-mobile/firefox-android/assets/92760693/f6547812-8c5f-4d54-876d-e89b9e3c1112" width="300">

The design also doesn't have top spacing between the label and the top of the screen we have all over the settings screens. I kept the spacing for consistency.

Reply:
+1 with keeping consistent with W400 for now. Let's file a followup bug to see if we should make sure all the other headline should be using W500.